docs/control-structures.rst

```diff
@@ -443,6 +370,9 @@ In any case, you will get a warning about the outer variable being shadowed.
         }
     }
 
+.. warning::
+    Before version 0.5.0 Solidity followed the same scoping rules as JavaScript.
```

Comment:
Could you extend this section a little? Most people think that JavaScript has C99 scoping rules.